server.js

```diff
@@ -1598,6 +1598,25 @@ module.exports.startServer = async () => {
     throw new Error('unknown serverType: ' + config.serverType);
   }
 
+  // Wait for the server to either start successfully or error out.
+  await new Promise((resolve, reject) => {
```

Comment:
It's up to us whether we want to actually leave this in. This was useful for identifying the root cause of a test failure where multiple servers were trying to listed on the same port.

Reply:
Looks fine to me to keep.